doc/options.md

```diff
@@ -0,0 +1,2865 @@
+
+
+---
+
+### `mkProject.<in>.readTheDocs`
+
+**Type**: submodule
+
+**Default**: 
+```nix
+{
+  siteFolder = null;
+}
+```
+
+
+**Example**: 
+```nix
+# outputs.nix 
+{ repoRoot, inputs, pkgs, lib, system }:
+let 
+  cabalProject = lib.iogx.mkProject {
+    readTheDocs.siteFolder = "doc/read-the-docs-site";
+  };
+in 
+[
+  {
+    inherit cabalProject;
+  }
+  {
+    packages.read-the-docs-site = cabalProject.iogx.read-the-docs-site;
+  }
+]
+
+```
+
+
+Configuration for your [`read-the-docs`](https://readthedocs.org) site. 
+
+If no site is required, this option can be omitted.
+
+The shells generated by [`mkProject.<in>.mkShell`](#mkprojectinmkshell) will be augmented with several scripts to make developing your site easier, grouped under the tag `read-the-docs`.
+
+In addition, the `read-the-docs-site` derivation will be available in [`mkProject.<out>.iogx.read-the-docs-site`](#mkprojectoutiogxread-the-docs-site).
+
+
+---
+
+### `mkProject.<in>.readTheDocs.siteFolder`
+
+**Type**: null or string
+
+**Default**: `null`
+
+
+**Example**: 
+```nix
+# project.nix
+{ repoRoot, inputs, pkgs, lib, system }:
+
+lib.iogx.mkProject {
+  readTheDocs.siteFolder = "./doc/read-the-docs-site";
+}
+
+```
+
+
+A Nix string representing a path, relative to the repository root, to your site folder containing the `conf.py` file.
+
+If no site is required you can set this field to `null`, or omit the `readTheDocs` option entirely. 
+
+
+---
+
+### `mkProject.<out>.iogx`
+
+**Type**: submodule
+
+
+
+**Example**: 
+```nix
+# flake.nix 
+{
+  outputs = inputs: inputs.iogx.lib.mkFlake {
+    outputs = import ./outputs.nix;
+  };
+}
+
+# outputs.nix
+{ repoRoot, inputs, pkgs, lib, system }:
+let 
+  cabalProject = lib.iogx.mkProject {
+    cabalProjectArgs = {
+      compiler-nix-name = "ghc8107";
+
+      flake.variants.profiled = { 
+        modules = [{ enableProfiling = true; }];
+      };
+
+      flake.variants.ghc928 = { 
+        compiler-nix-name = "ghc928";
+      };
+    }
+  };
+in 
+[
+  {
+    inherit cabalProject;
+  }
+  {
+    pacakges.read-the-docs-site-ghc8107 = cabalProject.iogx.read-the-docs-site;
+    pacakges.read-the-docs-site-ghc928 = cabalProject.projectVariants.ghc928.iogx.read-the-docs-site;
+  }
+  {
+    devShells.default = cabalProject.iogx.devShell;
+    devShells.profiled = cabalProject.projectVariants.profiled.iogx.devShell;
+    devShells.ghc928 = cabalProject.projectVariants.ghc928.iogx.devShell;
+  }
+  {
+    hydraJobs.ghc8107 = cabalProject.iogx.hydraJobs;
+    hydraJobs.ghc928 = cabalProject.projectVariants.ghc928.iogx.hydraJobs;
+  }
+]
+
+```
+
+
+This in an attrset containing all the derivations for your project.
+
+Note that the `iogx` attrset will be avaialable for each of your project variants defined in [`mkProject.<in>.cabalProjectArgs`](#mkprojectincabalprojectargs).
+
+You will want to consume `iogx` in your flake [`mkFlake.<in>.outputs`](#mkflakeinoutputs), as shown in the example above.
```

Comment:
This seems like the exact reverse of the design we'd discussed: Instead of iogx providing a way to put the flake together with you providing individual bits, instead iogx expects you to put the flake together from individual bits it provides.